src/main/java/com/Incubatormsrv/Incubatormsrv/weather/service/WeatherService.java

```diff
@@ -0,0 +1,50 @@
+package com.Incubatormsrv.Incubatormsrv.weather.service;
+
+import _Users_luyanda_glucode.com_Documents_GitHub_Incubator_msrv.api.WeatherApi;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+@Component
+public class WeatherService {
```

Comment:
What you're defining here is baically bean configurations. 

This class should be dependent on the WeatherAPI provided by your retrofit as a bean to dependency inject.

Then this class should invoke the service call and handle the errors.